ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/PreExecutingSubmissionValidator.scala

```diff
@@ -133,18 +129,18 @@ class PreExecutingSubmissionValidator[WriteSet](
 
   private[preexecution] def generateReadSet(
       fetchedInputs: DamlStateMapWithFingerprints,
-      accessedKeys: Set[DamlStateKey]): ReadSet =
-    accessedKeys
-      .map { key =>
-        val (_, fingerprint) = fetchedInputs.getOrElse(
-          key,
-          throw new IllegalStateException(
-            "Committer accessed key that was not present in the input"))
-        key -> fingerprint
-      }
-      .map {
-        case (damlKey, fingerprint) =>
-          keySerializationStrategy.serializeStateKey(damlKey) -> fingerprint
+      accessedKeys: Set[DamlStateKey],
+  ): ReadSet =
+    accessedKeys.view
+      .flatMap { key =>
+        // If the key is missing, ignore it. The transaction should have been rejected anyway.
+        fetchedInputs
+          .get(key)
+          .map {
+            case (_, fingerprint) =>
+              keySerializationStrategy.serializeStateKey(key) -> fingerprint
+          }
+          .toList
       }
       .toVector
       .sortBy(_._1.asReadOnlyByteBuffer)
```

Comment:
Looks like this change probably isn't appropriate; the tests are unhappy so I guess it was there for a good reason. Not sure what to do here though. @miklos-da, do you have any thoughts?

Should we be getting the fingerprints of everything in the pre-execution result's read set, even if they're not actually touched in the case of rejection?

Reply:
Yes, all the accessed keys' fingerprints should be there for conflict detection, excluding the ones we didn't look at because of an early stop.
What specific type of key is missing in the input list but is being accessed?

Reply:
This is the error we're observing in _CommandDeduplicationIT_. A key was pointing at contract _A_ during pre-execution, but is now pointing at contract _B_. Therefore, contract _A_ is not in `fetchedInputs`.

The transaction will also be rejected because of stale inputs.